app/services/platforms/kick.ts

```diff
@@ -39,89 +66,259 @@ export class KickService
   implements IPlatformService {
   static initialState: IKickServiceState = {
     ...BasePlatformService.initialState,
-    settings: { title: '', streamUrl: '', streamKey: '' },
+    settings: {
+      title: '',
+      display: 'horizontal',
+      mode: 'landscape',
+    },
+    broadcastId: '',
+    ingest: '',
+    chatUrl: '',
+    channelName: '',
   };
 
-  searchGames?: (searchString: string) => Promise<IGame[]>;
-  scheduleStream?: (startTime: number, info: TStartStreamOptions) => Promise<any>;
-  getHeaders: (req: IPlatformRequest, useToken?: string | boolean) => Dictionary<string>;
-  streamPageUrl: string;
-  widgetsWhitelist?: WidgetType[];
+  @Inject() windowsService: WindowsService;
+  @Inject() diagnosticsService: DiagnosticsService;
 
   readonly apiBase = '';
+  readonly domain = 'https://kick.com';
   readonly platform = 'kick';
   readonly displayName = 'Kick';
-  readonly capabilities = new Set<TPlatformCapability>(['resolutionPreset']);
+  readonly capabilities = new Set<TPlatformCapability>(['title', 'viewerCount']);
 
-  readonly authWindowOptions = {};
-  readonly authUrl = '';
+  authWindowOptions: Electron.BrowserWindowConstructorOptions = {
+    width: 600,
+    height: 800,
+  };
 
-  fetchNewToken() {
-    return Promise.resolve();
+  private get oauthToken() {
+    return this.userService.views.state.auth?.platforms?.kick?.token;
   }
 
-  protected init() {
-    this.syncSettingsWithLocalStorage();
-
-    this.userService.userLogout.subscribe(() => {
-      this.updateSettings({ title: this.state.settings.title, streamUrl: '', streamKey: '' });
-    });
+  /**
+   * Kick's API currently does not provide viewer count.
+   * To prevent errors, return 0 for now;
+   */
+  get viewersCount(): number {
+    return 0;
```

Comment:
should we hide the viewercount from livedock if they're on kick only maybe?

Reply:
Good call